### PR TITLE
Clarify that messages starting a saga must have a mapping

### PR DESCRIPTION
**Pull request by andreasohlund**

### CODE REVIEW

File: nservicebus/sagas/index.md [hunk 1]
```diff
@@ -156,4 +156,4 @@ Sagas manage state of potentially long-running business processes. To access the
...
  * The saga data may not contain all the required data. A saga handling the order process may keep track of the "payment id" and the status of the payment, but it is not interested in keeping track of the amount paid. On the other hand, for querying it may be required to query the paid amount along with other data.
 
-The recommended approach is for the saga to publish events, containing the required data, and have handlers that process these events and store the data in one or more read model(s) for querying purposes. It reduces coupling to the internals of the specific saga persistence, removes contention and doesn't bypass the safeguard of the existing saga logic.
\ No newline at end of file
+The recommended approach is for the saga to publish events, containing the required data, and have handlers that process these events and store the data in one or more read model(s) for querying purposes. It reduces coupling to the internals of the specific saga persistence, removes contention and doesn't bypass the safeguard of the existing saga logic.
```
Comment (SimonCropp) [inline nservicebus/sagas/index.md]: What safeguards?

Comment (andreasohlund) [inline nservicebus/sagas/index.md]: Strange, I didn't change that sentence, can we deal with this in a new pull?

**Comment by andreasohlund** [comment]
@SimonCropp added one more commit

**Comment by dsaf** [comment]
> v6 will enforce that all messages starting a saga has a mapping (throw an exception if no mapping is found)

Does it mean that current v6 documentation sample is outdated? I cannot see the mapping for ```StartOrder``` message:

```csharp
public class OrderSaga : Saga<OrderSagaData>,
                        IAmStartedByMessages<StartOrder>,
                        IHandleMessages<CompleteOrder>
{
    protected override void ConfigureHowToFindSaga(SagaPropertyMapper<OrderSagaData> mapper)
    {
        mapper.ConfigureMapping<CompleteOrder>(message => message.OrderId)
                .ToSaga(sagaData => sagaData.OrderId);
    }

//...
}
```

> In short: Mappings are optional for messages that doesn't start the saga.

It's because in my current project all saga messages are correlated that I started assuming it's expected. Even ```Bus.Reply``` messages are mapped. For me it's a choice between having less code and making code more maintainable. Someone might: add a new message contract, but forget about correlation; change ```Reply``` to ```SendLocal``` but forget about correlation etc.

**Comment by andreasohlund** [comment]
Yep, sample is outdated, I will fix that!

>For me it's a choice between having less code and making code more maintainable. Someone might: add a new message contract, but forget about correlation; change Reply to SendLocal but forget about correlation etc.

It depends sometimes you can choose to not have a property to correlate to and let nservicebus autocorrelate the response for you:

http://docs.particular.net/nservicebus/sagas/#correlating-messages-to-a-saga-auto-correlation

so that is really up to you to make a tradeoff btw less code (no property on the response message + no mapping) and explicit code.

**Comment by andreasohlund** [comment]
Snippets fixed to map the start message as well https://github.com/Particular/docs.particular.net/pull/1463/commits/c5adc5018b294ef79c9d8c730726f581a687408b

**Comment by dsaf** [comment]
Thank you.

**Comment by andreasohlund** [comment]
@dsaf thanks again for the awesome feedback, it's really invaluable for us to be able to improve out doco so please keep it coming!

**Pull request merged**